packages/core/vue/src/useWallet.ts

```diff
@@ -109,36 +104,29 @@ export const createWalletStore = ({
 
     // Create a wallet dictionary keyed by their name.
     const walletsByName = computed(() => {
-        return wallets.reduce<Record<WalletName, Wallet>>((walletsByName, wallet) => {
+        return wallets.reduce<Record<WalletName, Adapter>>((walletsByName, wallet) => {
             walletsByName[wallet.name] = wallet;
             return walletsByName;
         }, {});
     });
 
-    // Update the wallet and adapter based on the wallet provider.
+    // Update the wallet based on the wallet provider.
     watch(
         name,
         (): void => {
             const wallet = walletsByName.value?.[name.value as WalletName] ?? null;
-            const adapter = wallet && wallet.adapter;
-            if (adapter) {
+            if (wallet) {
                 setState({
                     wallet,
-                    adapter,
-                    ready: false,
-                    publicKey: adapter.publicKey,
-                    connected: adapter.connected,
+                    ready: wallet.readyState == WalletReadyState.Installed,
+                    publicKey: wallet.publicKey,
+                    connected: wallet.connected,
+
                 });
 
-                // Asynchronously update the ready state
-                const waiting = name;
-                (async function () {
-                    const readyValue = await adapter.ready();
-                    // If the selected wallet hasn't changed while waiting, update the ready state
-                    if (name === waiting) {
-                        ready.value = readyValue;
-                    }
-                })();
+                wallet.on("readyStateChange", () => {
```

Comment:
Do these need to be unregistered if the adapters change?

Reply:
Yeah they should be, I'll make the change

Reply:
This has been moved to the 'watchEffect' call